design/ceph/object/ceph-sse-s3.md

```diff
@@ -0,0 +1,44 @@
+---
+# AWS server side encryption SSE-S3 support for RGW
+target-version: release-1.10
+---
+
+# Feature Name
+AWS server side encryption SSE-S3 support for RGW
+
+## Summary
+The S3 protocol over three different types of [server side encryption](https://docs.aws.amazon.com/AmazonS3/latest/userguide/serv-side-encryption.html): SSE-C, SSE-KMS and SSE-S3. For the last two RGW server need to configure with external services such as `vault`. Currently Rook can configure RGW with `SSE-KMS` options, so that S3 request with `sse:kms` header will be handled properly. Recently the support for handling the `sse:s3` added to RGW with another set of options, so Rook needs to provide option to configure RGW with those options.
+
+### Goals
+Configure RGW with `SSE-S3` options, so that RGW can handle request with `sse:s3` headers.
+ 
+### Non-Goals
+Both `SSE-KMS` and `SSE-S3` will be configured with same set of values such as `vault-server`, `vault-secret-engine` etc
+
+## Proposal details
+Introducing new field `ServerSideEncryptionSpec` in `CephObjectStoreSpec` which defines types of `AWS-SSE` configured with RGW.
+```
+type AWSServerSideEncryptionSpec {
+	S3Enabled bool `json:"s3Enabled,omitempty"`
+	KMSEnabled bool `json:"kmsEnabled,omitempty"`
+}
+type RGWServerSideEncryptionSpec struct {
+	AWS AWSServerSideEncryptionSpec `json:"aws,omitempty"`
+}
+type ObjectStoreSpec struct {
+...
+	ServerSideEncryption RGWServerSideEncryptionSpec `json:"serverSideEncryption,omitempty"`
+...
+} 
```

Comment:
Where are `vault-server` and `vault-secret-engine` in this spec?

If those are read from `spec.security`, then I think this needs to be a sub-section of `spec.security`. Otherwise, the API separates the options, suggesting they are not related.

Reply:
This discussion is cycling back again and again. Please see [comment](https://github.com/rook/rook/pull/10204#discussion_r876011585). `Security.kms` have connection details for the key management service required for both encryptions. Currently `sse:kms` is already part of Rook and it is enabled if the `Security` field basically contains key management service details. Since the option is specific to RGW or S3 it does not make sense to include in `Security` since it is more generic to Rook.Hence introduced new option in `ObjectStoreSpec`

Reply:
Could you also describe in the design spec why it needs to be separate from the `security` section?